packages/project/lib/specifications/types/Component.js

```diff
@@ -291,11 +291,21 @@ class Component extends ComponentProject {
 	}
 
 	async _ensureComponent() {
-		// Ensure that a Component.js exists
-		const componentResource = await this._getRawSourceReader().byPath("/Component.js");
-		if (!componentResource) {
+		// Throw if neither Component.js nor Component.ts is present
+		// or if both are present
+		const componentJSResource = await this._getRawSourceReader().byPath("/Component.js");
+		const componentTSResource = await this._getRawSourceReader().byPath("/Component.ts");
+		if (componentJSResource && componentTSResource) {
```

Comment:
I would expect this to be a real use case, where the output dir is the same as source dir.

In such case, IMO we should check only the TS file as it must be the source for the JS.

Let the others say what do they think about it